sdrf_pipelines/openms/openms.py

```diff
@@ -1,32 +1,63 @@
+r"""
+Convert SDRF files for use with OpenMS.
+
+Example command:
+
+parse_sdrf convert-openms \
+    -s .\sdrf-pipelines\sdrf_pipelines\large_sdrf.tsv \
+    -c '[characteristics[biological replicate],characteristics[individual]]'
+"""
+
 import logging
 import re
 from collections import Counter
 
 import pandas as pd
 
 from sdrf_pipelines.openms.unimod import UnimodDatabase
+from sdrf_pipelines.utils.utils import tsv_line
 
 logger = logging.getLogger(__name__)
 
-# example: parse_sdrf convert-openms -s .\sdrf-pipelines\sdrf_pipelines\large_sdrf.tsv -c '[characteristics[biological replicate],characteristics[individual]]'
-
 
 class FileToColumnEntries:
-    file2mods = {}
-    file2pctol = {}
-    file2pctolunit = {}
-    file2fragtol = {}
-    file2fragtolunit = {}
-    file2diss = {}
-    file2enzyme = {}
-    file2source = {}
-    file2label = {}
-    file2fraction = {}
-    file2combined_factors = {}
-    file2technical_rep = {}
-
-
-def get_openms_file_name(raw, extension_convert: str = None):
+    file2mods: dict[str, tuple] = {}
+    file2pctol: dict[str, str] = {}
+    file2pctolunit: dict[str, str] = {}
+    file2fragtol: dict[str, str] = {}
+    file2fragtolunit: dict[str, str] = {}
+    file2diss: dict[str, str] = {}
+    file2enzyme: dict[str, str] = {}
+    file2source: dict[str, str] = {}
+    file2label: dict[str, list[str]] = {}
+    file2fraction: dict[str, str] = {}
+    file2combined_factors: dict[str, str] = {}
+    file2technical_rep: dict[str, str] = {}
+
+
+def infer_tmtplex(label_set: set) -> str:
```

Comment:
Some users previously suggested it would be best to provide a label parameter to allow specifying the label type, which is a good idea. Because sometimes we cannot correctly infer when key channels are missing.